tests/dwd/observations/test_api_data.py

```diff
@@ -158,33 +157,80 @@ def test_dwd_observation_data_time_input():
         end_date=pd.Timestamp("1971-01-01"),
     )
 
+    with pytest.raises(StartDateEndDateError):
+        DWDObservationData(
+            station_ids=[1],
+            parameters=[DWDObservationParameterSet.CLIMATE_SUMMARY],
+            resolution=DWDObservationResolution.DAILY,
+            start_date="1971-01-01",
+            end_date="1951-01-01",
+        )
+
+
+def test_dwd_observation_data_dynamic_period():
+    # Historical period expected
     request = DWDObservationData(
         station_ids=[1],
         parameters=[DWDObservationParameterSet.CLIMATE_SUMMARY],
         resolution=DWDObservationResolution.DAILY,
-        periods=[DWDObservationPeriod.HISTORICAL],
         start_date="1971-01-01",
     )
 
-    assert request == DWDObservationData(
+    assert request.periods == [
+        DWDObservationPeriod.HISTORICAL,
+    ]
+
+    # Historical and recent period expected
+    request = DWDObservationData(
         station_ids=[1],
         parameters=[DWDObservationParameterSet.CLIMATE_SUMMARY],
         resolution=DWDObservationResolution.DAILY,
-        periods=[
-            DWDObservationPeriod.HISTORICAL,
-        ],
-        start_date=pd.Timestamp("1971-01-01"),
-        end_date=pd.Timestamp("1971-01-01"),
+        start_date="1971-01-01",
+        end_date=pd.Timestamp(datetime.now()) - pd.Timedelta(days=400),
     )
 
-    with pytest.raises(StartDateEndDateError):
-        DWDObservationData(
-            station_ids=[1],
-            parameters=[DWDObservationParameterSet.CLIMATE_SUMMARY],
-            resolution=DWDObservationResolution.DAILY,
-            start_date="1971-01-01",
-            end_date="1951-01-01",
-        )
+    assert request.periods == [
+        DWDObservationPeriod.HISTORICAL,
+        DWDObservationPeriod.RECENT,
+    ]
+
+    # Historical, recent and now period expected
+    request = DWDObservationData(
+        station_ids=[1],
+        parameters=[DWDObservationParameterSet.CLIMATE_SUMMARY],
+        resolution=DWDObservationResolution.DAILY,
+        start_date="1971-01-01",
+        end_date=pd.Timestamp(datetime.now()),
+    )
+
+    assert request.periods == [
+        DWDObservationPeriod.HISTORICAL,
+        DWDObservationPeriod.RECENT,
+        DWDObservationPeriod.NOW,
+    ]
+
+    # !!!Recent and now period cant be tested dynamically
+    # TODO: add test with mocked datetime here
```

Comment:
We might want to use [`freezegun`](https://pypi.org/project/freezegun/) for that.

Reply:
Unfortunately freezegun isn't doing well with dateparser so I'd leave it for now and maybe open another issue touching dateparsing in general.